src/plugins/dataflow/model/dataflow-content.ts

```diff
@@ -205,6 +223,39 @@ export const DataflowContentModel = TileContentModel
       if (!foundFlag){
         self.dataSet.removeAttribute(attribute);
       }
+    },
+    addLinkedTable(tableId: string) {  //tableID is table we linked it to
+      const sharedModelManager = self.tileEnv?.sharedModelManager;
+      if (sharedModelManager?.isReady && !self.isLinkedToTable(tableId)) {
+        const sharedTable = sharedModelManager.findFirstSharedModelByType(SharedDataSet, tableId);
+        const tableTile = getTileContentById(self, tableId); //get tableTile contents given a tableId
+        //sever connection table -> table sharedDataSet
+        sharedTable && sharedModelManager.removeTileSharedModel(tableTile, sharedTable);
+        //connect table -> dataflow sharedDataset
+        self.sharedModel && sharedModelManager.addTileSharedModel(tableTile, self.sharedModel);
+      }
+      else {
+        console.warn("GeometryContent.addLinkedTable unable to link table");
+      }
+    },
+    removeLinkedTable(tableId: string) {
+      const sharedModelManager = self.tileEnv?.sharedModelManager;
+      if (sharedModelManager?.isReady && self.isLinkedToTable(tableId)) {
+        const sharedTable = sharedModelManager.findFirstSharedModelByType(SharedDataSet, tableId);
```

Comment:
If DataFlow is linked to the table then that means the table's dataset should be same as DataFlow's dataset. So that is the dataSet that should be used. 

`findFirstSharedModelByType(SharedDataSet, tableId)` will be looking for a shared model in the whole document that has a `providerId` of `tableId`.  The providerId of dataSets that are created by DataFlow should be set to DataFlow not to the Table. So this call should not actually find anything. If it is finding something it means the providerId is not be set right.
So it shouldn't be necessary to look up the sharedTable like this